kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenProjectLoader.java

```diff
@@ -88,24 +89,32 @@ public static MavenProject parseMavenPom(InputStream pomStream, boolean offline)
     public static MavenEmbedder newMavenEmbedder(boolean offline) {
         MavenRequest mavenRequest = createMavenRequest(offline);
         MavenEmbedder mavenEmbedder;
-            try {
-                mavenEmbedder = new MavenEmbedder( mavenRequest );
-            } catch (MavenEmbedderException e) {
-                log.error("Unable to create new MavenEmbedder", e);
-                throw new RuntimeException(e);
-            }
+        try {
+            mavenEmbedder = new MavenEmbedder( mavenRequest );
+        } catch (MavenEmbedderException e) {
+            log.error("Unable to create new MavenEmbedder", e);
+            throw new RuntimeException(e);
+        }
         return mavenEmbedder;
     }
 
+    public static String getGlobalRepo(){
+        String global = System.getProperty(GLOBAL_M2_REPO_URL);
+        if(global == null){
+            global = "repositories" +File.separator +"kie" +File.separator +"global";
+        }
+        return global;
+    }
+
     public static MavenRequest createMavenRequest(boolean offline) {
         MavenRequest mavenRequest = new MavenRequest();
-        mavenRequest.setLocalRepositoryPath( MavenSettings.getSettings().getLocalRepository() );
-        mavenRequest.setUserSettingsSource(MavenSettings.getUserSettingsSource());
+        mavenRequest.setLocalRepositoryPath(getGlobalRepo());
+        //mavenRequest.setUserSettingsSource(MavenSettings.getUserSettingsSource());
 
         // BZ-1007894: If dependency is not resolvable and maven project builder does not complain about it,
         // then a <code>java.lang.NullPointerException</code> is thrown to the client.
         // So, the user will se an exception message "null", not descriptive about the real error.
-        mavenRequest.setResolveDependencies( true );
+        //mavenRequest.setResolveDependencies(false);//offline works with false
```

Comment:
please max, stop keep commented code in PRs =/

Reply:
Fixed